c/ring_statistics.cpp

```diff
@@ -141,7 +141,7 @@ py_distances_on_graph(PyObject *self, PyObject *args)
     int nat = *std::max_element(i, i+nneigh)+1;
 
     /* Construct seed array */
-    int seed[nat+1];
+    int* seed = new int[nat+1];
```

Comment:
I think this leaks memory, since `delete[] seed;` is missing. I'd suggest `std::vector` since it follows RAII.